DemoApp/app/src/main/java/com/colortv/demo/ExoPlayerVideoActivity.java

```diff
@@ -0,0 +1,195 @@
+package com.colortv.demo;
+
+import android.app.Activity;
+import android.net.Uri;
+import android.os.Bundle;
+import android.os.Handler;
+import android.util.Log;
+
+import com.colortv.android.ColorTvContentRecommendationListener;
+import com.colortv.android.ColorTvError;
+import com.colortv.android.ColorTvSdk;
+import com.google.android.exoplayer2.DefaultLoadControl;
+import com.google.android.exoplayer2.ExoPlaybackException;
+import com.google.android.exoplayer2.ExoPlayer;
+import com.google.android.exoplayer2.ExoPlayerFactory;
+import com.google.android.exoplayer2.LoadControl;
+import com.google.android.exoplayer2.SimpleExoPlayer;
+import com.google.android.exoplayer2.Timeline;
+import com.google.android.exoplayer2.extractor.DefaultExtractorsFactory;
+import com.google.android.exoplayer2.extractor.ExtractorsFactory;
+import com.google.android.exoplayer2.source.ExtractorMediaSource;
+import com.google.android.exoplayer2.source.MediaSource;
+import com.google.android.exoplayer2.source.TrackGroupArray;
+import com.google.android.exoplayer2.trackselection.AdaptiveVideoTrackSelection;
+import com.google.android.exoplayer2.trackselection.DefaultTrackSelector;
+import com.google.android.exoplayer2.trackselection.TrackSelection;
+import com.google.android.exoplayer2.trackselection.TrackSelectionArray;
+import com.google.android.exoplayer2.trackselection.TrackSelector;
+import com.google.android.exoplayer2.ui.SimpleExoPlayerView;
+import com.google.android.exoplayer2.upstream.BandwidthMeter;
+import com.google.android.exoplayer2.upstream.DataSource;
+import com.google.android.exoplayer2.upstream.DefaultBandwidthMeter;
+import com.google.android.exoplayer2.upstream.DefaultDataSourceFactory;
+import com.google.android.exoplayer2.util.Util;
+
+public class ExoPlayerVideoActivity extends Activity {
+
+    public static final String VIDEO_URL = "videoUrl";
+    public static final String VIDEO_ID = "videoId";
+    private final String PLACEMENT = "TestContent";
+    private String videoId;
+    private boolean showContentRecActionInvoked = false;
+    private boolean contentRecommendationLoaded = false;
+    private SimpleExoPlayer simpleExoPlayer;
+    private SimpleExoPlayerView simpleExoPlayerView;
+
+    private ColorTvContentRecommendationListener loadShowContentRecommendationListener = new ColorTvContentRecommendationListener() {
```

Comment:
This is all overcomplicated. There should only be one listener in the main activity, which should launch this activity for result, get the result and based on this decide if it should display content recommendation. That way the code is much simpler and more readable for demonstration purposes.